src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,142 @@
 import React from 'react';
+import cn from 'classnames';
 import './TodoList.scss';
+import { getTodosFromServer, getTodosByStatus } from '../../api/loadData';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+type Props = {
+  setUserId: (userId: number) => void,
+};
+
+type State = {
+  todos: Todo[],
+  searchInput: string,
+  status: string,
+};
+
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    todos: [],
+    searchInput: '',
+    status: 'all',
+  };
+
+  async componentDidMount() {
+    const todos = await getTodosFromServer();
+
+    this.setState({
+      todos: [...todos],
+    });
+  }
+
+  componentDidUpdate(_: {}, prevState: State) {
+    const { status } = this.state;
+
+    if (status !== prevState.status) {
+      this.loadTodosByStatus();
+    }
+  }
+
+  getTodosBySearch = (todos: Todo[], searchInput: string): Todo[] => {
+    return todos.filter(todo => (
+      todo.title.toLowerCase().includes(searchInput.toLowerCase())
```

Comment:
make a separate variable for `searchInput.toLowerCase()` before filtering, no need to do this operation on every iteration